### DISCUSSION TITLE
FundMe.sol Question

### DISCUSSION THREAD

**Discussion by DeanKrupka** [Cyfrin/foundry-full-course-cu#1082] [Q&A] [answered]
Hello. Just wondering why in FundMe.sol (Remix lesson), Patrick wrote the following:
'''
    function fund() public payable {
        require(msg.value.getConversionRate() >= MINIMUM_USD, "You need to spend more ETH!");
        // require(PriceConverter.getConversionRate(msg.value) >= MINIMUM_USD, "You need to spend more ETH!");
        addressToAmountFunded[msg.sender] += msg.value;
        funders.push(msg.sender);
    }
    '''
Wondering if anyone can explain why he was able to write the require statement the way he did, rather than formatting it as the commented line.

**Comment by sunnyStefi** [comment, marked as answer]
Hi! 
Both statements are valid. The`PriceConverter` library has been imported inside `FundMe.sol` and the directive `using PriceConverter for uint256;` has been declared at the begin of the contract (line 10).

This directive attaches the functions of `PriceConverter` to the type uint256 and a **function overload** takes place: the function `getConversionRate()` belongs to the uint256 type from now on (only inside this contract) .

`msg.sender` is then used as first parameter `ethAmount` for `getConversionRate`.

**Reply by DeanKrupka** [reply]
Thanks for the response! Got it.  

ps. I think you mean msg.value (not msg.sender) in last statement.

**Reply by sunnyStefi** [reply]
Yes exactly! Have an awesome coding journey!
